library/libosbuild/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+name = "libosbuild"
+version = "73.0.0"
+edition = "2021"
+authors = ["supakeen <supakeen@redhat.com>"]
+description = "`libosbuild` library modules."
+
+[dependencies]
+walkdir = "^2.3.2"
```

Comment:
You might consider https://docs.rs/cap-std/ - on the ostree side we've made an investment in that ecosystem and it's a very well designed library that has a pretty good match with the problem domain of operating system builds, where you have *two* root filesystems.  rpm-ostree uses it extensively too.

I maintain https://docs.rs/cap-std-ext/latest/cap_std_ext/ which is a set of high level helpers on top, including things like an `atomic_write` API that properly uses Linux `O_TMPFILE` etc.

Reply:
(The only problem I keep hitting is that cap-std doesn't want to follow/read *absolute* symlinks; so you need to drop down a level for that.  See e.g. https://github.com/bytecodealliance/cap-std/issues/251 )

Reply:
I gladly support that recommendation of `cap-std` (and capability-based logic in general). And cap-std is supported by the bytecodealliance (org behind wasm) these days, so the maintenance is solid.

Reply:
That sounds (much) better, `walkdir` is what I found initially when looking for an alternative of `std::io::fs::walk_dir`. I'll build on top of that.